dotCMS/src/main/java/com/dotcms/publishing/job/SiteSearchJobImpl.java

```diff
@@ -129,182 +134,213 @@ public void run(final JobExecutionContext jobContext) throws JobExecutionExcepti
+         }
+
+         Logger.info(SiteSearchJobImpl.class, () -> String
+                 .format("Incremental mode [%s]. current index is `%s`. new index is `%s`. alias is `%s`  bundle id is `%s` ",
+                         BooleanUtils.toStringYesNo(incremental), indexName ,
+                         UtilMethods.isSet(newIndexName) ? newIndexName : "N/A",
+                         indexAlias,
+                         bundleId)
+                 );
+
+         final List<Host> hosts;
+         if (indexAll) {
+             hosts = hostAPI.findAll(userToRun, true);
+         } else {
+             hosts = Stream.of(indexHosts).map(h -> {
+                 try {
+                     return hostAPI.find(h, userToRun, true);
+                 } catch (DotDataException | DotSecurityException e) {
+                     Logger.error(SiteSearchJobImpl.class, e);
+                 }
+                 return null;
+             }).filter(Objects::nonNull).collect(Collectors.toList());
+         }
+
+         final Builder<SiteSearchConfig> builder = ImmutableList.builder();
+
+         final List<String> languageToIndex = Arrays.asList((String[])dataMap.get(LANG_TO_INDEX));
+         final ListIterator<String> listIterator = languageToIndex.listIterator();
+         while (listIterator.hasNext()) {
+             final String lang = listIterator.next();
+             final SiteSearchConfig config = new SiteSearchConfig();
+             config.setJobId(jobId);
+             config.setLanguage(Long.parseLong(lang));
+             config.setJobName(jobName);
+             config.setHosts(hosts);
+             config.setNewIndexName(newIndexName);
+             config.setIndexName(indexName);
+             config.setIndexAlias(indexAlias);
+             config.setId(bundleId);
+             config.setStartDate(startDate);
+             config.setEndDate(endDate);
+             config.setIncremental(incremental);
+             config.setUser(userToRun);
+
+             if(include) {
+                 config.setIncludePatterns(paths);
+             } else {
+                 config.setExcludePatterns(paths);
+             }
+
+             //We should always replace the index when performing on non-incremental mode.
+             //That means we drop the old one and re-use the alias.
+             //But we only activate the new index when the old one was the default.
+             //Or there wasn't any previous index.
+             //it must be done on the last round of our loop.
+             final boolean switchIndex = !incremental && !listIterator.hasNext();
+             config.setSwitchIndexWhenDone(switchIndex);
+             builder.add(config);
+         }
+         final String joinedHosts = UtilMethods.join(indexHosts,",",true);
+         final String langList = UtilMethods.join(languageToIndex, ",");
+         final String pathsAsString = paths.size() > 0 ? UtilMethods.join(paths,",") : "/*";
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Substitute calls to size() == 0 (or size() != 0, size() > 0, size() < 1) with calls to isEmpty()](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5188880)